### PR TITLE
fix: validate action of getSignedUrl() function

### LINKED ISSUE #671: getSignedUrl fails silently with invalid url if `action` is invalid/undefined
The `file.getSignedUrl` method will successfully return a `signedUrl` but when making a request via the URL will give a `SignatureDoesNotMatch` response (expected).

- [ ] Add validation for the `action` parameter.

### PR DESCRIPTION
**Pull request by AVaksman**
Fixes #671

- [x] Tests and linter pass
- [x] Code coverage does not decrease

add validation for `action` parameter.

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -2319,6 +2319,12 @@ class File extends ServiceObject<File> {
    */
   getSignedUrl(cfg: GetSignedUrlConfig, callback?: GetSignedUrlCallback):
       void|Promise<GetSignedUrlResponse> {
+    const validAction = ['read', 'write', 'delete', 'resumable'];
```
Comment (bcoe) [inline src/file.ts]: This looks reasonable to me, might be worth considering using an `Enum` for `TypeScript`, does this translate cleanly to the JavaScript API?

Comment (stephenplusplus) [reply src/file.ts]: There's actually already this enum: https://github.com/googleapis/nodejs-storage/blob/7b1ef2c376e353758879050107426a980c502774/src/file.ts#L2341

We could just validate that `method` is something.

Comment (AVaksman) [reply src/file.ts]: Thanks, done

**Comment by AVaksman** [comment]
> What was the behavior of the old API, if I were to pass in an invalid action? This looks great, my only concern is it might be a breaking change:
> 
> * if the old logic would have failed with a more cryptic error, this seems reasonable as a non-breaking change.
> * if prior to this we would have succeeded, this feels like a breaking change.
> 
> Assuming we would have previously been throwing an exception, I'm 👍 on this change.

Previously `getSignedUrl` with invalid `action` parameter would still successfully return an invalid URL (a URL with permissions for non existing action so to speak) and any request via this URL would result with `SignatureDoesNotMatch` response. The error is cryptic since it points user in completely different direction.

**Comment by bcoe** [comment]
@AVaksman feels like a `fix` to me, I continue to be a :+1:

**Pull request merged**